src/main/java/org/dependencytrack/persistence/ProjectQueryManager.java

```diff
@@ -88,6 +88,11 @@ public PaginatedResult getProjects(final boolean includeMetrics, final boolean e
         var filterBuilder = new ProjectQueryFilterBuilder()
                 .excludeInactive(excludeInactive);
 
+        if (onlyRoot){
+            filterBuilder.excludeProjectsThatAreChildren();
+            pm.getFetchPlan().addGroup(Project.FetchGroup.ALL.name());
```

Comment:
The fetch plan should be set for `query`, otherwise it may have unintended impact on other queries performed by this `pm`.